### PR TITLE
🐛 Bug fix 828 a few of the docs-ecutable docs are finishing with errors

### PR DESCRIPTION
**Pull request by fileppb**
## Summary
1) Fix the docs-ecutable so it runs from the current github branch environment rather than cloning a previously packaged release.
Otherwise the newly pushed code is never exercised in the case of contributions.
2) Fix the docs-ecutable related to issue 814, pivoting from kcp's synctargets.workload.kcp.io and locations.scheduling.kcp.io to kubestellar's synctargets.edge.kcp.io and locations.edge.kcp.io in order to help wean kubestellar from kcp.

Comment: Unfortunately this PR has many, many commits. These commits were used to test in the Github Actions environment, since I thought there might be a timing issue involved.

## Related issue(s)
814
828
Fixes #
828

### CODE REVIEW

**Comment by fileppb** [comment]
Oh...the business about the customizer, and loc-name and env replacements is STILL not resolved. But it seems important to get the CI workflows functional again.

File: docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-post-kcp.md [hunk 1]
```diff
@@ -32,21 +43,45 @@ The kubectl plugin lines use fully specific executables (e.g.,
...
+are namespaced and are meaningful in KubeStellar.
+
+```shell
+kubestellar init
+```
+
+### Create an inventory management workspace.
+```shell
+kubectl ws root
+kubectl ws create imw-1 --enter
```
Comment (MikeSpreitzer) [inline docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-post-kcp.md:65]: The `--enter` is not needed, since the next command sets the currrent workspace.

File: docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-post-kcp.md [hunk 2]
```diff
@@ -1,25 +1,36 @@
 <!--example1-post-kcp-start-->
-### Create an inventory management workspace.
+### Use the root workspace.
```
Comment (MikeSpreitzer) [inline docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-post-kcp.md:2]: This section now accomplishes nothing.

File: docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-stage-1b.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ write a file containing YAML for deploying the syncer in the guilder
 cluster.
 
 ```shell
+kubectl kcp bind apiexport root:espw:edge.kcp.io
```
Comment (MikeSpreitzer) [inline docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-stage-1b.md:10]: This command should not be needed, the `prep-for-syncer` plugin is not sensitive to the current working directory when you specify the IMW to be something other than `.`.

File: docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-stage-2.md [hunk 4]
```diff
@@ -82,7 +82,7 @@ data:
     <html>
       <body>
         This is a common web site.
-        Running in %(loc-name).
+        Running in %(env).
```
Comment (MikeSpreitzer) [inline docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-stage-2.md:85]: Why this change?

File: docs/content/Coding Milestones/PoC2023q1/kubestellar-syncer-subs/kubestellar-syncer-0-deploy-guilder.md [hunk 5]
```diff
@@ -2,7 +2,11 @@
 Go to inventory management workspace and find the mailbox workspace name.
 ```shell
 kubectl ws root:imw-1
-mbws=`kubectl get SyncTarget guilder -o jsonpath="{.metadata.annotations['kcp\.io/cluster']}-mb-{.metadata.uid}"`
+kubectl get SyncTargets
+kubectl get synctargets.edge.kcp.io
+kubectl describe synctargets.edge.kcp.io guilder
+#kubectl describe Synctarget guilder
```
Comment (MikeSpreitzer) [inline docs/content/Coding Milestones/PoC2023q1/kubestellar-syncer-subs/kubestellar-syncer-0-deploy-guilder.md:8]: This is script debugging that should eventually be removed.  Not necessarily in this PR.

File: docs/content/Coding Milestones/PoC2023q1/mailbox-controller.md [hunk 6]
```diff
@@ -135,13 +135,14 @@ In a separate terminal window(3), create an inventory management workspace as fo
 ```shell
 kubectl ws \~
 kubectl ws create imw --enter
+kubectl kcp bind apiexport root:espw:edge.kcp.io
```
Comment (MikeSpreitzer) [inline docs/content/Coding Milestones/PoC2023q1/mailbox-controller.md:138]: This added command should also get added text talking about it.

File: docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-stage-1b.md (hunk 3 above)
Comment (fileppb) [reply docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-stage-1b.md:10]: probably true, this was desperate groping before the revelation that the docs-ecutable wa;t actually testing the current branch

File: docs/content/Coding Milestones/PoC2023q1/mailbox-controller.md (hunk 6 above)
Comment (fileppb) [reply docs/content/Coding Milestones/PoC2023q1/mailbox-controller.md:138]: ok

File: docs/content/Coding Milestones/PoC2023q1/kubestellar-syncer-subs/kubestellar-syncer-0-deploy-guilder.md (hunk 5 above)
Comment (fileppb) [reply docs/content/Coding Milestones/PoC2023q1/kubestellar-syncer-subs/kubestellar-syncer-0-deploy-guilder.md:8]: ok

File: docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-stage-2.md (hunk 4 above)
Comment (fileppb) [reply docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-stage-2.md:85]: it was a test that accomplished nothing, I still don't know what's wrong with the customization/variable relplaccement

File: docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-post-kcp.md (hunk 2 above)
Comment (fileppb) [reply docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-post-kcp.md:2]: true

File: docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-post-kcp.md (hunk 1 above)
Comment (fileppb) [reply docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-post-kcp.md:65]: was simply copied from previous md

File: docs/content/Coding Milestones/PoC2023q1/outline.md [hunk 7]
```diff
@@ -153,8 +153,8 @@ in an eventually consistent way, it is not just one-and-done.
 
 In this design the primary interface between infrastructure management
 and workload management is API objects in _inventory management_
-workspaces.  We abuse the `Location` and `SyncTarget` object types
-from [kcp TMC](https://github.com/kcp-dev/kcp/tree/v0.11.0/pkg/apis) for
+workspaces.  We use the `Location` and `SyncTarget` object types
+from Kubestellar's edge.kcp.io api group (created by `kubestellar init`) for
```
Comment (MikeSpreitzer) [inline docs/content/Coding Milestones/PoC2023q1/outline.md:157]: The api group is not "created" by `kubestellar init`.
Those object types are not "created" by `kubestellar init`.
The `kubestellar init` command makes the KubeStellar Core Space export those resources.

File: docs/content/Coding Milestones/PoC2023q1/where-resolver-subs/where-resolver-1-build-kubestellar.md [hunk 8]
```diff
@@ -1,5 +1,9 @@
 <!--where-resolver-1-build-kubestellar-start-->
 ```shell
+make imports
+make codegen
+make crds
+make update-contextual-logging
```
Comment (MikeSpreitzer) [inline docs/content/Coding Milestones/PoC2023q1/where-resolver-subs/where-resolver-1-build-kubestellar.md:6]: These do not belong here.
These affect files that are maintained in git.
Developers making relevant changes are responsible getting these done.

File: docs/content/Coding Milestones/PoC2023q1/where-resolver.md [hunk 9]
```diff
@@ -107,6 +107,7 @@ I0605 10:53:00.261128   29786 controller.go:201] "starting controller" controlle
 Use workspace `root:compute` as the Inventory Management Workspace (IMW).
 ```shell
 kubectl ws root:compute
+kubectl kcp bind apiexport root:espw:edge.kcp.io
```
Comment (MikeSpreitzer) [inline docs/content/Coding Milestones/PoC2023q1/where-resolver.md:110]: This command does not belong here, unless I have overlooked something.

File: docs/content/Coding Milestones/PoC2023q1/outline.md (hunk 7 above)
Comment (fileppb) [reply docs/content/Coding Milestones/PoC2023q1/outline.md:157]: ok, yes, make sense

File: docs/content/Coding Milestones/PoC2023q1/where-resolver-subs/where-resolver-1-build-kubestellar.md (hunk 8 above)
Comment (fileppb) [reply docs/content/Coding Milestones/PoC2023q1/where-resolver-subs/where-resolver-1-build-kubestellar.md:6]: yes and no. 1) Is there anywhere in the documentation that instructs contributors to take those steps before commiting their changes? maybe I forgot that's in the instructions1 or didn't see it? I do it, but that's because of discussions with our team. 2) does it hurt?

File: docs/content/Coding Milestones/PoC2023q1/where-resolver.md (hunk 9 above)
Comment (fileppb) [reply docs/content/Coding Milestones/PoC2023q1/where-resolver.md:110]: probably a residual from desperate attempts to deal with the fundamental problem of cloning previously packaged release and not working from the current contribution. But I'm reluctant to fix what seems to work now,

File: docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-stage-2.md (hunk 4 above)
Comment (MikeSpreitzer) [reply docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-stage-2.md:85]: You will want to remove this change, it is not helping.

File: docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-post-kcp.md [hunk 10]
```diff
@@ -32,21 +43,45 @@ The kubectl plugin lines use fully specific executables (e.g.,
...
 
 Use the following two commands. They label both florin and guilder
 with `env=prod`, and also label guilder with `extended=si`.
 
 ```shell
+kubectl ws root:imw-1
 kubectl kubestellar ensure location florin  loc-name=florin  env=prod
 kubectl kubestellar ensure location guilder loc-name=guilder env=prod extended=si
+echo "decribe the florin location object"
+kubectl describe location.edge.kcp.io florin
```
Comment (MikeSpreitzer) [inline docs/content/Coding Milestones/PoC2023q1/example1-subs/example1-post-kcp.md:77]: `get -o yaml` would be better than`describe` here.  There are no associated events (which `describe` would display if there were any), and the Managed Fields (which are displayed by `describe` but not normally by `get`) are verbose and not valuable here.

**Comment by MikeSpreitzer** [comment]
The reason that customization is not working is that some edits were overlooked in #814 .  The rubber meets the road at https://github.com/fileppb/edge-mc/blob/new-issue-559-560/pkg/placement/workload-projector.go#L1091

File: docs/content/Coding Milestones/PoC2023q1/where-resolver.md (hunk 9 above)
Comment (MikeSpreitzer) [reply docs/content/Coding Milestones/PoC2023q1/where-resolver.md:110]: Yes, I tested without this addition and got as much success as with it.

File: docs/content/Coding Milestones/PoC2023q1/where-resolver-subs/where-resolver-1-build-kubestellar.md (hunk 8 above)
Comment (MikeSpreitzer) [reply docs/content/Coding Milestones/PoC2023q1/where-resolver-subs/where-resolver-1-build-kubestellar.md:6]: We lack proper documentation how to build.  The proposed solution in #821 calls for that.

Having these commands here does not cause the execution to fail, but it does lead human readers in the wrong direction.

**Comment by MikeSpreitzer** [comment]
/lgtm

This is good enough to merge, it is progress and the remaining issues can be addressed in follow-up PRs.  However, it would be better if all the changes in the PR were squashed into one commit.  Better, but I will not insist on it.  Let me know how you want to proceed.

**Comment by fileppb** [comment]
well I've never squashed before and after my last experience screwing up a re-base, I'm reluctant to experiment and learn on something that is in the critical path for Tuesday's release. But I absolutely agree that this PR is ugly and has too many commits.

**Comment by fileppb** [comment]
Cancelling this PR and substituting with PR #835 https://github.com/kubestellar/kubestellar/pull/835
